src/lib/homestore_backend/replication_state_machine.cpp

```diff
@@ -159,30 +164,148 @@ ReplicationStateMachine::create_snapshot(std::shared_ptr< homestore::snapshot_co
     auto ctx = dynamic_pointer_cast< homestore::nuraft_snapshot_context >(context);
     auto s = ctx->nuraft_snapshot();
     LOGI("create snapshot, last_log_idx_: {} , last_log_term_: {}", s->get_last_log_idx(), s->get_last_log_term());
+    m_snapshot_context = context;
     return folly::makeSemiFuture< homestore::ReplResult< folly::Unit > >(folly::Unit{});
 }
 
 bool ReplicationStateMachine::apply_snapshot(std::shared_ptr< homestore::snapshot_context > context) {
-    LOGE("apply_snapshot not implemented");
-    return false;
+    // TODO persist snapshot
+    m_snapshot_context = context;
+    return true;
 }
 
-std::shared_ptr< homestore::snapshot_context > ReplicationStateMachine::last_snapshot() {
-    LOGE("last_snapshot not implemented");
-    return nullptr;
-}
+std::shared_ptr< homestore::snapshot_context > ReplicationStateMachine::last_snapshot() { return m_snapshot_context; }
 
 int ReplicationStateMachine::read_snapshot_data(std::shared_ptr< homestore::snapshot_context > context,
```

Comment:
Not sure if it is too late/worth...

the naming `read_snapshot_data` is conflict with this 
 interface, which is misleading.

https://github.com/eBay/NuRaft/blob/99eeef34a2620686e0dd40ad7fbd5cab561140fc/include/libnuraft/state_machine.hxx#L235-L255

Reply:
Looks like its deprecated and nuraft_mesg is not using it too. And HO is not including it directly.

Reply:
yes by conflict I mean, it is easy to be confused :).   I was wrongly assume we are using the deprecated endpoint then later I aware we translate that in homestore .

Reply:
Yeah didnt know this name was used. We can rename it in future, as it needs changes in HS.